utils/params.go

```diff
@@ -791,28 +792,54 @@ func readConfigFromTarget(client vcsclient.VcsClient, gitParamsFromEnv *Git) (co
 	return
 }
 
-// This function fetches a config profile if JF_CONFIG_PROFILE is provided.
-// If so - it verifies there is only a single module with a '.' path from root. If these conditions doesn't hold we return an error.
-func getConfigProfileIfExistsAndValid(xrayVersion, xscVersion string, jfrogServer *coreconfig.ServerDetails) (configProfile *services.ConfigProfile, err error) {
+// This function attempts to fetch a config profile if JF_USE_CONFIG_PROFILE is set to true.
+// If we need to use a profile, we first try to get the profile by name that can be provided through JF_CONFIG_PROFILE. If name is provided but profile doesn't exist we return an error.
+// If we need to use a profile, but name is not provided, we check if there is a config profile associated to the repo URL.
+// When a profile is found we verify several conditions on it.
+// If a profile was requested but not found by url nor by name we return an error.
+func getConfigProfileIfExistsAndValid(xrayVersion, xscVersion string, jfrogServer *coreconfig.ServerDetails, gitClient vcsclient.VcsClient, gitParams *Git) (configProfile *services.ConfigProfile, err error) {
+	useConfigProfile := strings.ToLower(getTrimmedEnv(JfrogUseConfigProfileEnv))
+	if useConfigProfile == "false" || useConfigProfile == "" {
+		log.Debug(fmt.Sprintf("Configuration Profile usage is disabled. All configurations will be derived from environment variables and files.\nTo enable a Configuration Profile, please set %s to TRUE", JfrogUseConfigProfileEnv))
+		return
+	}
+
+	// Attempt to get the config profile by profile's name
 	profileName := getTrimmedEnv(JfrogConfigProfileEnv)
-	if profileName == "" {
-		log.Debug(fmt.Sprintf("No %s environment variable was provided. All configurations will be induced from Env vars and files", JfrogConfigProfileEnv))
+	if profileName != "" {
+		log.Debug(fmt.Sprintf("Configuration profile was requested. Searching profile by provided name '%s'", profileName))
+		if configProfile, err = xsc.GetConfigProfileByName(xrayVersion, xscVersion, jfrogServer, profileName); err != nil || configProfile == nil {
+			return
+		}
+		err = verifyConfigProfileValidity(configProfile)
 		return
 	}
 
-	if configProfile, err = xsc.GetConfigProfile(xrayVersion, xscVersion, jfrogServer, profileName); err != nil {
+	// Getting repository's url in order to get repository HTTP url
+	repositoryInfo, err := gitClient.GetRepositoryInfo(context.Background(), gitParams.RepoOwner, gitParams.RepoName)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I don't know if this is a good place to handle this, it will cause a multiple duplicate API calls to fetch this info, since this is called for every command and we already calling it for each command:
* https://github.com/jfrog/frogbot/blob/c300767011d099a0fc479e73daf5c8e8270196c6/scanpullrequest/scanpullrequest.go#L253
* https://github.com/jfrog/frogbot/blob/c300767011d099a0fc479e73daf5c8e8270196c6/scanrepository/scanrepository.go#L140